app/Http/Controllers/RefreshedCollectionController.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Http\Controllers;
+
+use App\Jobs\FetchCollectionFloorPrice;
+use App\Jobs\FetchCollectionOwners;
+use App\Jobs\FetchCollectionTraits;
+use App\Jobs\FetchCollectionVolume;
+use Illuminate\Http\JsonResponse;
+use Illuminate\Http\RedirectResponse;
+use Illuminate\Http\Request;
+use Illuminate\Support\Facades\Bus;
+
+class RefreshedCollectionController extends Controller
+{
+    public function show(Request $request): JsonResponse
+    {
+        return response()->json([
+            'indexing' => $request->wallet()->is_refreshing_collections,
+        ]);
+    }
+
+    public function store(Request $request): RedirectResponse
+    {
+        $wallet = $request->wallet();
+
+        abort_unless($wallet->canRefreshCollections(), 403);
+
+        $collections = $wallet->nfts->load('collection.network')->map->collection->flatten()->unique();
+
+        $wallet->update([
+            'is_refreshing_collections' => true,
+        ]);
+
+        Bus::batch($collections->flatMap(fn ($collection) => [
+            new FetchCollectionFloorPrice($collection->network->chain_id, $collection->address),
+            new FetchCollectionTraits($collection),
+            new FetchCollectionOwners($collection),
+            new FetchCollectionVolume($collection),
```

Comment:
I think it makes sense to somehow impose a global rate limit here, not just per wallet because there can be overlap in the collections between wallets.

While we already should have limits on each individual collection job, having a dedicated job makes it more flexible and extensible later on. It also improves observability, since we can correlate jobs more easily.

So maybe we should just wrap the individual dispatches in a dedicated job like `RefreshCollection` and this job then can be rate limited similar to the `RefreshNftMetadata` one. Just pick any sensible default.

You can check how it's done here:

https://github.com/ArdentHQ/dashbrd/blob/ee4f2ac5e16b60d71345c79aaca76f5dc002b1d2/app/Http/Controllers/RefreshedNftController.php#L22

https://github.com/ArdentHQ/dashbrd/blob/ee4f2ac5e16b60d71345c79aaca76f5dc002b1d2/app/Providers/AppServiceProvider.php#L67

The cooldown on the wallet, etc. can stay as is I think.